src/viam/ur/module/ur_arm.cpp

```diff
@@ -383,10 +384,14 @@ void URArm::move_through_joint_positions(const std::vector<std::vector<double>>&
     // TODO: use options
     if (!positions.empty()) {
         std::vector<Eigen::VectorXd> waypoints;
+
         for (auto position : positions) {
             const Eigen::VectorXd next_waypoint_deg =
                 Eigen::VectorXd::Map(position.data(), boost::numeric_cast<Eigen::Index>(position.size()));
             const Eigen::VectorXd next_waypoint_rad = next_waypoint_deg * (M_PI / 180.0);  // convert from radians to degrees
+            if ((!waypoints.empty()) && (next_waypoint_rad.isApprox(waypoints.back()))) {
```

Comment:
What's the epsilon here for `isApprox`? I think we should put it in a named constant, with units. I think these are radians so `k_waypoint_equivalancy_epsilon_rad` or something.